packages/cli/lib/services/response-saver.service.ts

```diff
@@ -60,35 +83,99 @@ export function onAxiosRequestFulfilled({
         saveResponse<Pick<Connection, 'metadata' | 'connection_config'>>({
             directoryName,
             data: { metadata: connection.metadata as Metadata, connection_config: connection.connection_config },
-            customFilePath: 'mocks/nango/getConnection.json',
-            concatenateIfExists: false
+            customFilePath: 'mocks/nango/getConnection.json'
         });
 
         saveResponse<Metadata>({
             directoryName,
             data: connection.metadata as Metadata,
-            customFilePath: 'mocks/nango/getMetadata.json',
-            concatenateIfExists: false
+            customFilePath: 'mocks/nango/getMetadata.json'
         });
 
         return response;
     }
 
-    const [pathname, params] = response.request.path.split('?');
-    const strippedPath = pathname.replace('/', '');
-
-    let concatenateIfExists = false;
-
-    if (params && params.includes('page')) {
-        concatenateIfExists = true;
-    }
+    const requestIdentity = computeConfigIdentity(response.config);
 
-    saveResponse<AxiosResponse>({
+    saveResponse<CachedRequest>({
         directoryName,
-        data: response.data,
-        customFilePath: `mocks/nango/${method}/${strippedPath}/${syncName}.json`,
-        concatenateIfExists
+        data: {
+            ...requestIdentity,
+            response: response.data,
+            status: response.status,
+            headers: response.headers as Record<string, string>
+        },
+        customFilePath: `mocks/nango/${requestIdentity.method}/proxy/${requestIdentity.endpoint}/${syncName}/${requestIdentity.requestIdentityHash}.json`
     });
 
     return response;
 }
+
+function computeConfigIdentity(config: AxiosRequestConfig): ConfigIdentity {
```

Comment:
I'm a bit confused by this, what does config identity means?

Reply:
It's basically a computed identity for a request config for the case of our integration tests in `integration-templates`. We'd consider two requests identical if the JSON.stringy of two ConfigIdentity objects is the same.

Reply:
mmh yeah okay, in that case it's not really just an identity it's just an object that can be used as an identity, got confused by that and config is generic but I guess it maps axios `config`.

Reply:
Yeah, Config is kind of overloaded for sure. I'm dumping the object only so that it's easier for us to debug when identities don't match - it can be tricky with just a hash to know why something changed.